packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -207,4 +233,36 @@ export class FileNavigatorContribution extends AbstractViewContribution<FileNavi
             this.selectWidgetFileNode(this.shell.currentWidget);
         }
     }
+
+    protected removeFolderFromWorkspace(): void {
+        this.workspaceService.removeFolders(
+            this.getSelectedNodesInFileNavigator()
+                .filter(node => this.isRootDirectory(node))
+                .map(rootFolder => {
+                    const rootFolderNode = rootFolder as FileStatNode;
+                    return new URI(rootFolderNode.uri.toString());
+                })
+        );
+    }
+
+    protected isRootFolderSelected(): boolean {
+        return this.getSelectedNodesInFileNavigator().some(node => this.isRootDirectory(node));
+    }
+
+    private getSelectedNodesInFileNavigator(): Readonly<SelectableTreeNode>[] {
```

Comment:
```
protected async getSelectedNodes(): Promise<ReadonlyArray<Readonly<SelectableTreeNode>>> {
    const navigator = await this.widget;
    return navigator.model.selectedNodes;
}
```

please also default to `protected`, so bugs can be worked around by concrete products

Reply:
I thought about doing it this way however the `async` part made `isRootFolderSelected()` (a function that calls `getSelectedNodes()`) async as well, and thus became incompatible with `CommandHandler`

```ts
export interface CommandHandler {
    /**
     * Execute this handler.
     */
    execute(...args: any[]): any;
    /**
     * Test whether this handler is enabled (active).
     */
    isEnabled?(...args: any[]): boolean;
    /**
     * Test whether menu items for this handler should be visible.
     */
    isVisible?(...args: any[]): boolean;
    /**
     * Test whether menu items for this handler should be toggled.
     */
    isToggled?(...args: any[]): boolean;
}
```
where the return type of `isVisible()` is `boolean`